catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceCallable.java

```diff
@@ -243,48 +235,57 @@ public ReliableResourceStatus call() {
 
                 bytesRead.addAndGet(n);
 
-                if (cacheFileOutputStream != null) {
-                    try {
-                        cacheFileOutputStream.write(buffer, 0, n);
-                    } catch (IOException e) {
-                        LOGGER.info("IOException during write to cached file's OutputStream", e);
-                        reliableResourceStatus =
-                                new ReliableResourceStatus(DownloadStatus.CACHED_FILE_OUTPUT_STREAM_EXCEPTION,
-                                        bytesRead.get());
-                    }
-                }
-
-                if (countingFbos != null) {
-                    try {
-                        countingFbos.write(buffer, 0, n);
-                        countingFbos.flush();
-                    } catch (IOException e) {
-                        LOGGER.info(
-                                "IOException during write to FileBackedOutputStream for client to read",
-                                e);
-                        reliableResourceStatus =
-                                new ReliableResourceStatus(DownloadStatus.CLIENT_OUTPUT_STREAM_EXCEPTION,
-                                        bytesRead.get());
-                    }
-                }
+                tryWriteCacheFileOutputStream(buffer, n);
+                tryWriteCountingFbos(buffer, n);
 
-                // Return status here so that each stream can be attempted to be updated regardless of
-                // which one might have had an exception
+                // Attempt to update each stream regardless of which one might have had an exception
                 if (reliableResourceStatus != null) {
                     return reliableResourceStatus;
                 }
             }
             LOGGER.trace("chunkCount = {},  bytesRead = {}", chunkCount, bytesRead.get());
         }
 
+        checkForSuccessfulCompletion();
```

Comment:
Instead of having a `checkForSuccessfulCompletion()` and then returning `reliableResourceStatus`, should we do `return getResourceDownloadCompletionStatus()`? That would prevent this side-effect based assignment to the `reliableResourceStatus` member variable.